File: react-kanban/src/types/ui.ts
export type Color =
  | "none"
  | "primary"
  | "link"
  | "info"
  | "success"
  | "warning"
  | "danger"
  | "white"
  | "light"
  | "dark"
  | "black"
  | "text"
  | "ghost";

export type Size = "small" | "normal" | "medium" | "large";

export type ButtonState = "normal" | "loading" | "disabled";

export type Alignment = "left" | "right";

export type InputType = "text" | "date" | "email" | "password" | "number";

export const DEFAULT_COLOR: Color = "none";
export const DEFAULT_SIZE: Size = "normal";
export const DEFAULT_BUTTON_STATE: ButtonState = "normal";
export const DEFAULT_ALIGNMENT: Alignment = "left";
export const DEFAULT_INPUT_TYPE: InputType = "text";

export const colorClasses: Record<Color, string> = {
  none: "",
  primary: "is-primary",
  link: "is-link",
  info: "is-info",
  success: "is-success",
  warning: "is-warning",
  danger: "is-danger",
  white: "is-white",
  light: "is-light",
  dark: "is-dark",
  black: "is-black",
  text: "is-text",
  ghost: "is-ghost",
};

export const textColorClasses: Record<Color, string> = {
  none: "",
  primary: "has-text-primary",
  link: "has-text-link",
  info: "has-text-info",
  success: "has-text-success",
  warning: "has-text-warning",
  danger: "has-text-danger",
  white: "has-text-white",
  light: "has-text-light",
  dark: "has-text-dark",
  black: "has-text-black",
  text: "",
  ghost: "",
};

export const sizeClasses: Record<Size, string> = {
  small: "is-small",
  normal: "",
  medium: "is-medium",
  large: "is-large",
};

export const buttonStateClasses: Record<ButtonState, string> = {
  normal: "",
  loading: "is-loading",
  disabled: "",
};

export const alignmentClasses: Record<Alignment, string> = {
  left: "",
  right: "is-right",
};

export const inputHtmlTypes: Record<InputType, string> = {
  text: "text",
  date: "date",
  email: "email",
  password: "password",
  number: "number",
};

File: leptos-kanban/src/types/ui.rs
#[allow(dead_code)]
#[derive(Default, Debug, Clone)]
pub enum Color {
    #[default]
    None,
    Primary,
    Link,
    Info,
    Success,
    Warning,
    Danger,
    White,
    Light,
    Dark,
    Black,
    Text,
    Ghost,
}

impl Color {
    pub fn as_class(&self) -> &'static str {
        match self {
            Self::None => "",
            Self::Primary => "is-primary",
            Self::Link => "is-link",
            Self::Info => "is-info",
            Self::Success => "is-success",
            Self::Warning => "is-warning",
            Self::Danger => "is-danger",
            Self::White => "is-white",
            Self::Light => "is-light",
            Self::Dark => "is-dark",
            Self::Black => "is-black",
            Self::Text => "is-text",
            Self::Ghost => "is-ghost",
        }
    }

    pub fn as_text_color_class(&self) -> &'static str {
        match self {
            Self::None => "",
            Self::Primary => "has-text-primary",
            Self::Link => "has-text-link",
            Self::Info => "has-text-info",
            Self::Success => "has-text-success",
            Self::Warning => "has-text-warning",
            Self::Danger => "has-text-danger",
            Self::White => "has-text-white",
            Self::Light => "has-text-light",
            Self::Dark => "has-text-dark",
            Self::Black => "has-text-black",
            Self::Text => "",
            Self::Ghost => "",
        }
    }
}

#[allow(dead_code)]
#[derive(Default, Debug, Clone)]
pub enum Size {
    Small,
    #[default]
    Normal,
    Medium,
    Large,
}

impl Size {
    pub fn as_class(&self) -> &'static str {
        match self {
            Self::Small => "is-small",
            Self::Normal => "",
            Self::Medium => "is-medium",
            Self::Large => "is-large",
        }
    }
}

#[allow(dead_code)]
#[derive(Default, Debug, Clone)]
pub enum ButtonState {
    #[default]
    Normal,
    Loading,
    Disabled,
}

impl ButtonState {
    pub fn as_class(&self) -> &'static str {
        match self {
            ButtonState::Normal | ButtonState::Disabled => "",
            ButtonState::Loading => "is-loading",
        }
    }
}

#[allow(dead_code)]
#[derive(Default, Debug, Clone)]
pub enum Alignment {
    #[default]
    Left,
    Right,
}

impl Alignment {
    pub fn as_class(&self) -> &'static str {
        match self {
            Alignment::Left => "",
            Alignment::Right => "is-right",
        }
    }
}

#[allow(dead_code)]
#[derive(Default, Debug, Clone)]
pub enum InputType {
    #[default]
    Text,
    Date,
    Email,
    Password,
    Number,
}

impl InputType {
    pub fn as_html_type(&self) -> &'static str {
        match self {
            InputType::Text => "text",
            InputType::Date => "date",
            InputType::Email => "email",
            InputType::Password => "password",
            InputType::Number => "number",
        }
    }
}
